cmd/in_memory/channel_controller/main.go

```diff
@@ -20,13 +20,88 @@ import (
 	// Uncomment the following line to load the gcp plugin (only required to authenticate against GKE clusters).
 	// _ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
 
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
 	"knative.dev/pkg/injection/sharedmain"
+	"knative.dev/pkg/signals"
+	"knative.dev/pkg/webhook"
+	"knative.dev/pkg/webhook/certificates"
+	"knative.dev/pkg/webhook/resourcesemantics"
+	"knative.dev/pkg/webhook/resourcesemantics/defaulting"
+	"knative.dev/pkg/webhook/resourcesemantics/validation"
 
+	messagingv1 "knative.dev/eventing/pkg/apis/messaging/v1"
```

Comment:
is there a reason these two imports are in a separate group? what do you think of reordering to be the 3 groups: stdlib, 3rd party and knative?

Reply:
we import in this order:

1.  built in
2. external
3. package local

Reply:
I had that wrong then, thanks :+1:  I thought the third group was knative.